Chess-Challenge/src/Framework/Application/Core/ChallengeController.cs

```diff
@@ -282,6 +282,29 @@ void EndGame(GameResult result, bool log = true, bool autoStartNextBotMatch = tr
                     Log("Game Over: " + result, false, ConsoleColor.Blue);
                 }
 
+                if (PlayerWhite.IsBot || PlayerBlack.IsBot)
+                {
+                    // Call optional end-of-game method for training convenience
+                    try
+                    {
+                        if (PlayerWhite.IsBot)
+                        {
+                            API.Board botBoard = new(board);
+                            PlayerWhite.Bot.GameOver(botBoard);  
+                        }
+                        if (PlayerBlack.IsBot)
+                        {
+                            API.Board botBoard = new(board);
+                            PlayerBlack.Bot.GameOver(botBoard);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Log("An error occurred while invoking the GameOver method\n" + e.ToString(), true, ConsoleColor.Red);
+                    }
```

Comment:
Just stumbled across this, but isn't the if-check on line 285 redundant here since you check both cases anyways in the try-block?

```suggestion           
                // Call optional end-of-game method for training convenience
                try
                {
                    if (PlayerWhite.IsBot)
                    {
                        API.Board botBoard = new(board);
                        PlayerWhite.Bot.GameOver(botBoard);  
                    }
                    if (PlayerBlack.IsBot)
                    {
                        API.Board botBoard = new(board);
                        PlayerBlack.Bot.GameOver(botBoard);
                    }
                }
                catch (Exception e)
                {
                    Log("An error occurred while invoking the GameOver method\n" + e.ToString(), true, ConsoleColor.Red);
                 
```

Reply:
True. I wanted the outer if closure in case there was some code that needed to run before calling the GameOver functions. Your suggestion is an optimization in this case.